libwallet/assets/dcr/consensus.go

```diff
@@ -111,24 +110,21 @@ func (asset *Asset) SetVoteChoice(agendaID, choiceID, hash, passphrase string) e
 		return err
 	}
 
-	currentChoice := w.AgendaChoice{
-		AgendaID: agendaID,
-		ChoiceID: "abstain", // default to abstain as current choice if not found in wallet
+	currentChoice, ok := choices[agendaID]
+	if ok && currentChoice == strings.ToLower(choiceID) {
+		// Do not set the same choice again
+		return nil
 	}
```

Comment:
The purpose of this is to prevent unnecessary API calls. It does this by preventing the existing choice being set again.
I don't think setting the same choice again is necessary.